nameresolution/dns/dns.go

```diff
@@ -26,7 +27,11 @@ func (k *resolver) Init(metadata nameresolution.Metadata) error {
 	return nil
 }
 
-// ResolveID resolves name to address in orchestrator.
+// ResolveID resolves name to address use DNS
 func (k *resolver) ResolveID(req nameresolution.ResolveRequest) (string, error) {
-	return fmt.Sprintf("%s-dapr.%s.svc:%d", req.ID, req.Namespace, req.Port), nil
+	if strings.TrimSpace(req.Namespace) != "" {
+		return fmt.Sprintf("%s-dapr.%s.svc:%d", req.ID, req.Namespace, req.Port), nil
+	}
+
+	return fmt.Sprintf("%s-dapr:%d", req.ID, req.Port), nil
```

Comment:
Hi, @abogdanav37. I'm trying to understand the rationale for this change. For service to service invocation, Dapr apps are not aware of the namespace. Can you clarify the need for this DNS pattern? Thanks!

Reply:
@abogdanov37 just a friendly ping.

Reply:
Hi! Sorry for the long response time. The main idea is running in different environments. When we run in Kubernetes we can use namespace but when we run in docker-compose it won't work.

Reply:
We can use mDNS mode for this case but it works only on a single machine

Reply:
@pkedy @yaron2 Can this PR be merged?